src/schema/search.ts

```diff
@@ -1,11 +1,33 @@
 import { IResolvers } from '@graphql-tools/utils';
 import { traceResolvers } from './trace';
 import { Context } from '../Context';
-import { postFeedback, SearchResultFeedback } from '../integrations';
+import {
+  getSessions,
+  postFeedback,
+  SearchResultFeedback,
+  SearchSession,
+  SearchSessionParams,
+} from '../integrations';
 import { ValidationError } from 'apollo-server-errors';
 import { GQLEmptyResponse } from './common';
 
+type GQLSearchSession = Pick<SearchSession, 'id' | 'prompt' | 'createdAt'>;
+type GQLSearchSessionParams = Pick<SearchSessionParams, 'limit' | 'lastId'>;
+
 export const typeDefs = /* GraphQL */ `
+  type SearchSession {
+    id: String!
+    prompt: String!
+    createdAt: String!
+  }
+
+  extend type Query {
+    """
+    Get user's search history
+    """
+    searchSessionHistory(limit: Int, lastId: String): [SearchSession]! @auth
```

Comment:
Usually we return a pagination object for such cases. This is the same as reading history and so on

Reply:
Cool. Let me build a paginator for something outside the graphorm query.

Reply:
You can utilize the same components and make them generic :)

Reply:
Yes, I mean, I will use our existing building blocks but make it applicable to our new requirement.